tests/provider: Provider produced inconsistent result after apply (r/eks_cluster)

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

On GovCloud, and potentially other non-standard partitions, 9 acceptance tests fail:

```
TestAccAWSEksCluster_basic
TestAccAWSEksCluster_EncryptionConfig
TestAccAWSEksCluster_Logging
TestAccAWSEksCluster_Tags
TestAccAWSEksCluster_Version
TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess
TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess
TestAccAWSEksCluster_VpcConfig_PublicAccessCidrs
TestAccAWSEksCluster_VpcConfig_SecurityGroupIds
```

### Terraform CLI and Terraform AWS Provider Version

Terraform 0.12.29 (& 0.13.4)
AWS 3.10.0

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* aws_emr_cluster

### Terraform Configuration Files

```hcl
data "aws_availability_zones" "available" {
  state = "available"

  filter {
    name   = "opt-in-status"
    values = ["opt-in-not-required"]
  }
}

resource "aws_iam_role" "test" {
  name = "tf-acc-test"

  assume_role_policy = <<POLICY
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
        "Service": "eks.amazonaws.com"
      },
      "Action": "sts:AssumeRole"
    }
  ]
}
POLICY
}

resource "aws_iam_role_policy_attachment" "test-AmazonEKSClusterPolicy" {
  policy_arn = "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy"
  role       = aws_iam_role.test.name
}

resource "aws_vpc" "test" {
  cidr_block = "10.0.0.0/16"

  tags = {
    Name                          = "terraform-testacc-eks-cluster-base"
    "kubernetes.io/cluster/tfacc" = "shared"
  }
}

resource "aws_subnet" "test" {
  count = 2

  availability_zone = data.aws_availability_zones.available.names[count.index]
  cidr_block        = "10.0.${count.index}.0/24"
  vpc_id            = aws_vpc.test.id

  tags = {
    Name                          = "terraform-testacc-eks-cluster-base"
    "kubernetes.io/cluster/tfacc" = "shared"
  }
}

resource "aws_eks_cluster" "test" {
  name     = "tf-acc-test"
  role_arn = aws_iam_role.test.arn

  vpc_config {
    subnet_ids = aws_subnet.test[*].id
  }

  depends_on = [aws_iam_role_policy_attachment.test-AmazonEKSClusterPolicy]
}
```

### Debug Output

```
TestAccAWSEksCluster_basic: resource_aws_eks_cluster_test.go:76: Step 1/2 error: Error running apply: 2020/10/14 09:08:47 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0
        
        Error: Provider produced inconsistent result after apply
        
        When applying changes to
        aws_iam_role_policy_attachment.test-AmazonEKSClusterPolicy, provider
        "registry.terraform.io/-/aws" produced an unexpected new value for was
        present, but now absent.
        
        This is a bug in the provider, which should be reported in the provider's own
        issue tracker.
        
    TestAccAWSEksCluster_basic: testing_new.go:62: Error running post-test destroy, there may be dangling resources: 2020/10/14 09:08:51 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0
        
        Error: Error deleting IAM Role (tf-acc-test-sngud): DeleteConflict: Cannot delete entity, must detach all policies first.
        	status code: 409, request id: c32fec79-04a9-4358-8db8-b0255e6e80da
        
        
--- FAIL: TestAccAWSEksCluster_basic (49.90s)
```

### Expected Behavior

Passing tests

### Actual Behavior

Tests fail

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `TESTARGS="-run=TestAccAWSEksCluster_" make testacc`

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/aws/resource_aws_eks_cluster_test.go
+++ b/aws/resource_aws_eks_cluster_test.go
@@ -528,8 +528,10 @@ data "aws_availability_zones" "available" {
   }
 }
 
+data "aws_partition" "current" {}
+
 resource "aws_iam_role" "test" {
-  name = "%s"
+  name = %[1]q
 
   assume_role_policy = <<POLICY
 {
@@ -538,7 +540,7 @@ resource "aws_iam_role" "test" {
     {
       "Effect": "Allow",
       "Principal": {
-        "Service": "eks.amazonaws.com"
+        "Service": "eks.${data.aws_partition.current.dns_suffix}"
       },
       "Action": "sts:AssumeRole"
     }
@@ -548,7 +550,7 @@ POLICY
 }
 
 resource "aws_iam_role_policy_attachment" "test-AmazonEKSClusterPolicy" {
-  policy_arn = "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy"
+  policy_arn = "arn:${data.aws_partition.current.partition}:iam::aws:policy/AmazonEKSClusterPolicy"
   role       = aws_iam_role.test.name
 }
 
@@ -556,8 +558,8 @@ resource "aws_vpc" "test" {
   cidr_block = "10.0.0.0/16"
 
   tags = {
-    Name                       = "terraform-testacc-eks-cluster-base"
-    "kubernetes.io/cluster/%s" = "shared"
+    Name                          = "terraform-testacc-eks-cluster-base"
+    "kubernetes.io/cluster/%[1]s" = "shared"
   }
 }
 
@@ -569,19 +571,17 @@ resource "aws_subnet" "test" {
   vpc_id            = aws_vpc.test.id
 
   tags = {
-    Name                       = "terraform-testacc-eks-cluster-base"
-    "kubernetes.io/cluster/%s" = "shared"
+    Name                          = "terraform-testacc-eks-cluster-base"
+    "kubernetes.io/cluster/%[1]s" = "shared"
   }
 }
-`, rName, rName, rName)
+`, rName)
 }
 
 func testAccAWSEksClusterConfig_Required(rName string) string {
-	return fmt.Sprintf(`
-%s
-
+	return composeConfig(testAccAWSEksClusterConfig_Base(rName), fmt.Sprintf(`
 resource "aws_eks_cluster" "test" {
-  name     = "%s"
+  name     = %q
   role_arn = aws_iam_role.test.arn
 
   vpc_config {
@@ -590,17 +590,15 @@ resource "aws_eks_cluster" "test" {
 
   depends_on = [aws_iam_role_policy_attachment.test-AmazonEKSClusterPolicy]
 }
-`, testAccAWSEksClusterConfig_Base(rName), rName)
+`, rName))
 }
 
 func testAccAWSEksClusterConfig_Version(rName, version string) string {
-	return fmt.Sprintf(`
-%s
-
+	return composeConfig(testAccAWSEksClusterConfig_Base(rName), fmt.Sprintf(`
 resource "aws_eks_cluster" "test" {
-  name     = "%s"
+  name     = %q
   role_arn = aws_iam_role.test.arn
-  version  = "%s"
+  version  = %q
 
   vpc_config {
     subnet_ids = aws_subnet.test[*].id
@@ -608,15 +606,13 @@ resource "aws_eks_cluster" "test" {
 
   depends_on = [aws_iam_role_policy_attachment.test-AmazonEKSClusterPolicy]
 }
-`, testAccAWSEksClusterConfig_Base(rName), rName, version)
+`, rName, version))
 }
 
 func testAccAWSEksClusterConfig_Logging(rName string, logTypes []string) string {
-	return fmt.Sprintf(`
-%s
-
+	return composeConfig(testAccAWSEksClusterConfig_Base(rName), fmt.Sprintf(`
 resource "aws_eks_cluster" "test" {
-  name                      = "%s"
+  name                      = %q
   role_arn                  = aws_iam_role.test.arn
   enabled_cluster_log_types = ["%v"]
 
@@ -626,11 +622,11 @@ resource "aws_eks_cluster" "test" {
 
   depends_on = [aws_iam_role_policy_attachment.test-AmazonEKSClusterPolicy]
 }
-`, testAccAWSEksClusterConfig_Base(rName), rName, strings.Join(logTypes, "\", \""))
+`, rName, strings.Join(logTypes, "\", \"")))
 }
 
 func testAccAWSEksClusterConfigTags1(rName, tagKey1, tagValue1 string) string {
-	return testAccAWSEksClusterConfig_Base(rName) + fmt.Sprintf(`
+	return composeConfig(testAccAWSEksClusterConfig_Base(rName), fmt.Sprintf(`
 resource "aws_eks_cluster" "test" {
   name     = %[1]q
   role_arn = aws_iam_role.test.arn
@@ -645,11 +641,11 @@ resource "aws_eks_cluster" "test" {
 
   depends_on = [aws_iam_role_policy_attachment.test-AmazonEKSClusterPolicy]
 }
-`, rName, tagKey1, tagValue1)
+`, rName, tagKey1, tagValue1))
 }
 
 func testAccAWSEksClusterConfigTags2(rName, tagKey1, tagValue1, tagKey2, tagValue2 string) string {
-	return testAccAWSEksClusterConfig_Base(rName) + fmt.Sprintf(`
+	return composeConfig(testAccAWSEksClusterConfig_Base(rName), fmt.Sprintf(`
 resource "aws_eks_cluster" "test" {
   name     = %[1]q
   role_arn = aws_iam_role.test.arn
@@ -665,11 +661,11 @@ resource "aws_eks_cluster" "test" {
 
   depends_on = [aws_iam_role_policy_attachment.test-AmazonEKSClusterPolicy]
 }
-`, rName, tagKey1, tagValue1, tagKey2, tagValue2)
+`, rName, tagKey1, tagValue1, tagKey2, tagValue2))
 }
 
 func testAccAWSEksClusterConfig_EncryptionConfig(rName string) string {
-	return testAccAWSEksClusterConfig_Base(rName) + fmt.Sprintf(`
+	return composeConfig(testAccAWSEksClusterConfig_Base(rName), fmt.Sprintf(`
 resource "aws_kms_key" "test" {
   description             = %[1]q
   deletion_window_in_days = 7
@@ -693,13 +689,11 @@ resource "aws_eks_cluster" "test" {
 
   depends_on = [aws_iam_role_policy_attachment.test-AmazonEKSClusterPolicy]
 }
-`, rName)
+`, rName))
 }
 
 func testAccAWSEksClusterConfig_VpcConfig_SecurityGroupIds(rName string) string {
-	return fmt.Sprintf(`
-%s
-
+	return composeConfig(testAccAWSEksClusterConfig_Base(rName), fmt.Sprintf(`
 resource "aws_security_group" "test" {
   vpc_id = aws_vpc.test.id
 
@@ -709,7 +703,7 @@ resource "aws_security_group" "test" {
 }
 
 resource "aws_eks_cluster" "test" {
-  name     = "%s"
+  name     = %q
   role_arn = aws_iam_role.test.arn
 
   vpc_config {
@@ -719,53 +713,47 @@ resource "aws_eks_cluster" "test" {
 
   depends_on = [aws_iam_role_policy_attachment.test-AmazonEKSClusterPolicy]
 }
-`, testAccAWSEksClusterConfig_Base(rName), rName)
+`, rName))
 }
 
 func testAccAWSEksClusterConfig_VpcConfig_EndpointPrivateAccess(rName string, endpointPrivateAccess bool) string {
-	return fmt.Sprintf(`
-%[1]s
-
+	return composeConfig(testAccAWSEksClusterConfig_Base(rName), fmt.Sprintf(`
 resource "aws_eks_cluster" "test" {
-  name     = %[2]q
+  name     = %q
   role_arn = aws_iam_role.test.arn
 
   vpc_config {
-    endpoint_private_access = %[3]t
+    endpoint_private_access = %t
     endpoint_public_access  = true
     subnet_ids              = aws_subnet.test[*].id
   }
 
   depends_on = [aws_iam_role_policy_attachment.test-AmazonEKSClusterPolicy]
 }
-`, testAccAWSEksClusterConfig_Base(rName), rName, endpointPrivateAccess)
+`, rName, endpointPrivateAccess))
 }
 
 func testAccAWSEksClusterConfig_VpcConfig_EndpointPublicAccess(rName string, endpointPublicAccess bool) string {
-	return fmt.Sprintf(`
-%[1]s
-
+	return composeConfig(testAccAWSEksClusterConfig_Base(rName), fmt.Sprintf(`
 resource "aws_eks_cluster" "test" {
-  name     = %[2]q
+  name     = %q
   role_arn = aws_iam_role.test.arn
 
   vpc_config {
     endpoint_private_access = true
-    endpoint_public_access  = %[3]t
+    endpoint_public_access  = %t
     subnet_ids              = aws_subnet.test[*].id
   }
 
   depends_on = [aws_iam_role_policy_attachment.test-AmazonEKSClusterPolicy]
 }
-`, testAccAWSEksClusterConfig_Base(rName), rName, endpointPublicAccess)
+`, rName, endpointPublicAccess))
 }
 
 func testAccAWSEksClusterConfig_VpcConfig_PublicAccessCidrs(rName string, publicAccessCidr string) string {
-	return fmt.Sprintf(`
-%[1]s
-
+	return composeConfig(testAccAWSEksClusterConfig_Base(rName), fmt.Sprintf(`
 resource "aws_eks_cluster" "test" {
-  name     = %[2]q
+  name     = %q
   role_arn = aws_iam_role.test.arn
 
   vpc_config {
@@ -777,5 +765,5 @@ resource "aws_eks_cluster" "test" {
 
   depends_on = [aws_iam_role_policy_attachment.test-AmazonEKSClusterPolicy]
 }
-`, testAccAWSEksClusterConfig_Base(rName), rName, publicAccessCidr)
+`, rName, publicAccessCidr))
 }

